tools/compute_npi_reduction.py

```diff
@@ -0,0 +1,245 @@
+#############################################################################
+# Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+#
+# Authors: Martin J. Kuehn, David Kerkmann
+#
+# Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#############################################################################
+"""
+@file cumpute_npi_reduction.py
+
+@brief Computes the overall reduction in contacts by enforcing a NPI.
+"""
+
+import sys
+import numpy as np
+import os
+import pandas as pd
+from memilio.epidata import defaultDict as dd
+
+
+def find_county(county):
+    """"! Attempts to find a given county by name or ID and returns both if successful.
+
+    @param county Name or ID of county.
+    @return Name and ID of county.
+    """
+
+    if county == 'Germany':
+        return county, 0
+    else:
+        # try to find county ID
+        county_id = dd.invert_dict(dd.County).get(county, False)
+        # if county was not found, try to check if ID was passed instead
+        if not county_id:
+            county_id = county
+            county = dd.County.get(county_id)
+            if county:
+                print('Found county ' + county + ' with input ID ' + str(county_id) + '.')
+            else:
+                raise ValueError('County ' + str(county_id) + ' was not found in County list. Aborting.')
+    return county, county_id
+
+
+def get_data_path():
+    """"! Returns data path of memilio repository. Currently only works if repository is called 'memilio'.
+
+    @return Data path of memilio repository.
+    """
+
+    data_path = os.getcwd().split('memilio')[0] + 'memilio/data/'
+    #print("My current directory is : " + data_path)
+    return data_path
+
+
+def compute_population(county_id, filename):
+    """"! Computes current population across age groups from given county.
+
+    @param county_id ID of county.
+    @param filename Name of file which has the age-resolved data.
+    @return Array containing the population for each age group in the given county.
+    """
+
+    # TODO: We currently don't have age-resolved data for each county.
+    #       Once we have this data, find correct file and adapt extraction to that file format.
```

Comment:
I now use the age group transformation and the correct file. Would you prefer to use the download scripts here instead of using the already downloaded file?